### PR TITLE
Bfb 41

### PR DESCRIPTION
**Pull request by Gursaanj**
**JIRA Ticket**
https://perigongames.atlassian.net/browse/BFB-41

**Description**
Added Pause Menu into our scenes

Press Escape to open/close Pause Menu.

Resume,Reset and Quit works as intended! 
Will add Settings Functionality too after  #49 and #54 gets merged into develop!. 

PauseMenu

### CODE REVIEW

File: Assets/_BForBoss/Scripts/Managers/WorldManager.cs [hunk 1]
```diff
@@ -74,6 +93,21 @@ private void SetupLeaderboardViews()
             _perigonAnalytics.StartSession(SystemInfo.deviceUniqueIdentifier);
         }
 
+        private void Update()
+        {
+            if (Keyboard.current[PauseKey].wasPressedThisFrame)
```
Comment (Gursaanj) [inline Assets/_BForBoss/Scripts/Managers/WorldManager.cs:98]: Right now the InputSystem has the Escape key attached to the FirstPersonCharacter, where pressing it auto unlocks the cursor! So kinda makes unpausing a bit wonky here cause the cursor is supposed to lock but sometimes doesn't

If its cool @m4rtint can I remove the InputSystem for Escape?

Comment (m4rtint) [reply Assets/_BForBoss/Scripts/Managers/WorldManager.cs:98]: not yet, it might break some other stuff, i'll handle it when i connect stuff.

**Review by m4rtint — APPROVED** [review]
lgtm

**Review by m4rtint — APPROVED** [review]
it lgtm

**Pull request merged**